frontend/app/routes/_gcweb-app.view-letters._index.tsx

```diff
@@ -72,10 +70,25 @@ function LetterList({ letters }: LetterListProps) {
     });
   };
 
+  function formatDate(dateStr: string) {
+    const date = new Date(dateStr);
+    const year = date.getFullYear();
+    const month = (date.getMonth() + 1).toString().padStart(2, '0');
+    const day = date.getDate().toString().padStart(2, '0');
+
+    return `${year}-${month}-${day}`;
+  }
+
   return (
     <div>
-      <button onClick={() => setSortOrder(sortOrder === 'ASC' ? 'DESC' : 'ASC')}>Sort by date</button>
-      <div className="border-l border-r border-t">
+      <label htmlFor="sortOrder" className="text-sm">
+        <strong>{t('view-letters:index.filter')}</strong>
+      </label>
+      <select id="sortOrder" value={sortOrder} onChange={(e) => setSortOrder(e.target.value as 'ASC' | 'DESC')} className="text-sm">
```

Comment:
On select change event use `useNavigate()` hook to add search param `sort` with the select value. The loader will react on the new request and return the list sorted accordingly. If no `sort` pass to the loader then sort by `DESC` by default.